### PR TITLE
[#IOPID-2503] add alerts for download/delete failures

### PR DESCRIPTION
**Pull request by arcogabbo**
### Motivation and Context
add alerts for download/delete procedures failures

### CODE REVIEW

File: src/domains/functions/function_admin.tf [hunk 1]
```diff
@@ -299,6 +299,96 @@ module "function_admin_staging_slot" {
...
+    EOT
+
+  severity                = 1
+  auto_mitigation_enabled = false
+  location                = azurerm_resource_group.admin_rg.location
+
+  // check once every minute(evaluation_frequency)
+  // on the last minute of data(window_duration)
+  evaluation_frequency = "P1D"
+  window_duration      = "P1D"
```
Comment (gquadrati) [inline src/domains/functions/function_admin.tf:323]: Every minute or every day? Either fix the comment or the settings.
(In my opinion a daily check is enough)

File: src/domains/functions/function_admin.tf [hunk 2]
```diff
@@ -299,6 +299,96 @@ module "function_admin_staging_slot" {
...
+
+  criteria {
+    query                   = <<-QUERY
+exceptions
+| where customDimensions.name startswith "user.data.download"
+| where customDimensions.isReplay == true
```
Comment (gquadrati) [inline src/domains/functions/function_admin.tf:372]:
```suggestion
| where customDimensions.isReplay == false
```

File: src/domains/functions/function_admin.tf [hunk 3]
```diff
@@ -299,6 +299,96 @@ module "function_admin_staging_slot" {
...
+  // check once every minute(evaluation_frequency)
+  // on the last minute of data(window_duration)
+  evaluation_frequency = "P1D"
+  window_duration      = "P1D"
+
+  criteria {
+    query                   = <<-QUERY
+exceptions
+| where customDimensions.name startswith "user.data.delete"
+| where customDimensions.isReplay == true
```
Comment (gquadrati) [inline src/domains/functions/function_admin.tf:329]:
```suggestion
| where customDimensions.isReplay == false
```

File: src/domains/functions/function_admin.tf [hunk 4]
```diff
@@ -299,6 +299,96 @@ module "function_admin_staging_slot" {
...
+  location                = azurerm_resource_group.admin_rg.location
+
+  // check once every minute(evaluation_frequency)
+  // on the last minute of data(window_duration)
+  evaluation_frequency = "P1D"
+  window_duration      = "P1D"
+
+  criteria {
+    query                   = <<-QUERY
+exceptions
```
[the remark this answers is not in the document]
Comment (arcogabbo) [reply src/domains/functions/function_admin.tf:371]: thank you. done in https://github.com/pagopa/io-infra/pull/1334/commits/8371ddcb2dc3d010f49f5c9ff89b8bb95e33343f

File: src/domains/functions/function_admin.tf (hunk 1 above)
Comment (arcogabbo) [reply src/domains/functions/function_admin.tf]: changed comment in https://github.com/pagopa/io-infra/pull/1334/commits/8371ddcb2dc3d010f49f5c9ff89b8bb95e33343f

File: src/domains/functions/function_admin.tf (hunk 2 above)
Comment (arcogabbo) [reply src/domains/functions/function_admin.tf:374]: I intentionally put true to exclude false positives like, for example, whenever the orchestrator files a retry and that goes fine without any error.

Comment (gquadrati) [reply src/domains/functions/function_admin.tf:374]: We never raise an exception in this scenario.
And what if we add a new activity after this log? We'll have three logs, two of them with `isReplay=true`

File: src/domains/functions/function_admin.tf (hunk 3 above)
Comment (arcogabbo) [reply src/domains/functions/function_admin.tf]: I removed the filter to align the upstream implementation in https://github.com/pagopa/io-infra/pull/1334/commits/1636a27919134d045b53b406175c4f0c4adff880

File: src/domains/functions/function_admin.tf (hunk 2 above)
Comment (arcogabbo) [reply src/domains/functions/function_admin.tf]: you are right. however I removed the filter to align the upstream implementation in https://github.com/pagopa/io-infra/pull/1334/commits/1636a27919134d045b53b406175c4f0c4adff880

**Review by gquadrati — APPROVED** [review]

**Pull request merged**